pennylane/ops/qubit/matrix_ops.py

```diff
@@ -313,11 +352,39 @@ def compute_decomposition(D, wires):
 
         **Example:**
 
+        # TODO: Update once Identity has been replaced by GlobalPhase
         >>> qml.DiagonalQubitUnitary.compute_decomposition([1, 1], wires=0)
-        [QubitUnitary(array([[1, 0], [0, 1]]), wires=[0])]
+        [QubitUnitary(array([[0.36235775+0.93203909j, 0.        +0.j        ],
+                             [0.        +0.j        , 0.36235775+0.93203909j]]), wires=[0]),
+         RZ(1.5000000000000002, wires=[1]),
+         RZ(-0.10000000000000003, wires=[0]),
+         IsingZZ(0.2, wires=[0, 1])]
 
         """
-        return [QubitUnitary(DiagonalQubitUnitary.compute_matrix(D), wires=wires)]
+        n = len(wires)
+        phases = qml.math.real(qml.math.log(D) * (-1j))
+        coeffs = _walsh_hadamard_transform(phases, n).T
+        global_phase = qml.math.exp(1j * coeffs[0])
+        # For all other gates, there is a prefactor -1/2 to be compensated.
+        coeffs = coeffs * (-2.0)
+
+        ops = [QubitUnitary(qml.math.tensordot(global_phase, qml.math.eye(2), axes=0), wires[0])]
```

Comment:
We could use `s_prod` here. In my opinion it might make more sense to avoid using `QubitUnitary` in decompositions.

Reply:
Are we supporting the application of `s_prod` within circuits? :) 
I'm all in favor of not using `QubitUnitary`!
Another idea would be to add a global phase gate @trbromley

Reply:
That's a good point. I did use it for one of the decompositions for `qml.ctrl`. Now that I think about it, I don't know if that makes more or less sense than using `QubitUnitary`.

Reply:
That's a good point. I considered using `s_prod` in one of the `qml.ctrl` decompositions, but didn't end up using it. Now that I think about it, I'm not sure if it makes more or less sense to use `QubitUnitary` over `s_prod`.

Reply:
I'd go with `QubitUnitary` for now, then, if that works for you? :)